scale.lua

```diff
@@ -38,6 +38,9 @@ function Scale:__Initialize()
     self.scale = {}
     self.scale.x = nil
     self.scale.y = nil
+	self.from = {}
+	self.from.x = nil
+	self.from.y = nil
```

Comment:
```suggestion
    self.from = {}
    self.from.x = 1
    self.from.y = 1
```

Let's just store the default scalars here - it simplifies the logic later.